core-bundle/src/Image/Studio/Figure.php

```diff
@@ -326,7 +326,8 @@ static function (string $attribute, string $value) {
             $templateData['floatClass'] = " float_$floating";
         }
 
-        return $templateData;
+        // Add arbitrary template options
+        return array_merge($templateData, $this->getOptions());
```

Comment:
Should we allow recursive merging here? 
Like if someone wants to overwrite `['picture']['sources'][0]['data-custom-attr'] = 'foo'`?

Reply:
On the other hand, this would make it impossible to overwrite/remove properties that are arrays... 😕

Reply:
But it's me as developer who's calling `getLegacyTemplateData()` right? So I can merge that data with `getOptions()` myself the way I like if I want to?

Reply:
Right, but then you don’t need to use the options anyway I think.

The merging as it is now is probably correct, @m-vo you can resolve this conversation if you agree.

Reply:
There are two things to it (and maybe we should handle them separately):

1. I might want to pass things to my template with the figure. The figure macros for instance can put classes on the HTML elements or could in the future allow disabling the caption and so on. These are all things where the template decides what to do. The only thing that matters is the data being present :slightly_smiling_face:, so it would also be fine to set it under a key: `$templateData['options'] = $this->getOptions();`

    In all of my tests I never needed to add anything but classes which worked nicely (Twig). We could alter our PHP templates to support the same thing after the refactoring in #2072.

1. It's a bit cumbersome to merge data into PHP templates (e.g. `$template->setData(array_merge($template->getData(), $data))` and again different if a plain object is used as a base like it's done a lot in the legacy code. So you would probably use `applyLegacyTemplateData()` in most cases. If we want to allow setting/overwriting arbitrary values along we could add another argument `$additionalData` to this function and merge it before applying.

But maybe the current solution is good enough for both cases. In an ideal world you would not use the legacy strategies anyway and apply the figure directly to your template.

Reply:
> But maybe the current solution is good enough for both cases. 👍